src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter PATTERN = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder strB = new StringBuilder();
+        if ((names != null) && (data != null)) {
+            LocalDate dateLocalFrom = LocalDate.parse(dateFrom, PATTERN);
+            LocalDate dateLocalTo = LocalDate.parse(dateTo, PATTERN);
+            int[] salaries = new int[names.length];
+            for (String line : data) {
+                String[] strings = line.split(" ");
+                for (int j = 0; j < names.length; j++) {
+                    if (dateLocalFrom.compareTo(LocalDate.parse(strings[0], PATTERN)) < 1
+                            && dateLocalTo.compareTo(LocalDate.parse(strings[0], PATTERN)) > -1) {
+                        if (names[j].equals(strings[1])) {
+                            salaries[j] += Integer.parseInt(strings[2])
+                                    * Integer.parseInt(strings[3]);
+                        }
+                    }
+                }
+
+            }
+            strB.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+            for (int i = 0; i < names.length; i++) {
+                strB.append("\r\n").append(names[i]).append(" - ").append(salaries[i]);
```

Comment:
```suggestion
                strB.append(System.lineSeparator()).append(names[i]).append(DELIMITER).append(salaries[i]);
```

Reply:
Thanks for this: "System.lineSeparator()"!!! Now I understand why the build didn't pass

Reply:
GitHub runs your code on Linux, and you probably use Windows, new-line symbols are OS dependant, that's a good example for a famous "works on my machine" problem :) Be careful with file path separators and case sensitivity too